gapis/resolve/memory.go

```diff
@@ -127,17 +127,7 @@ func Memory(ctx context.Context, p *path.Memory, rc *path.ResolveConfig) (*servi
 				if rng.Overlaps(r) {
 					interval.Merge(&reads, rng.Window(r).Span(), false)
 					if p.IncludeTypes {
-						pathMemoryAsType := &path.MemoryAsType{
-							Address: rng.Base,
-							Size:    rng.Size,
-							Pool:    p.Pool,
-							After:   p.After,
-							Type: &path.Type{
-								TypeIndex: id,
-								API:       &path.API{ID: path.NewID(apiId)},
-							},
-						}
-						value, err := MemoryAsType(ctx, pathMemoryAsType, rc)
```

Comment:
This commit is undoing a lot of the work of the previous commit. It's very hard to know what the cumulative effect of these 5 commits is going to be once they are squished. Is there anything we can do to make this clearer?

Reply:
Hey Alan, I think the cumulative change given by github can make it clearer? (https://github.com/google/agi/pull/301/files)

Sorry for the deleting work in this commit, it was to remove the unnecessary path generation step, by directly transferring needed information.

Reply:
Hey, thanks Stella! Sorry, I was not aware of this github feature. It makes reviews like this much easier to read, thanks :)